### PR TITLE
KHP3-793 Add CaCx MOH 745 Report in KenyaEMR

### PR DESCRIPTION
**Pull request by CynthiaKamau**
moh columns

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        cd.setDescription("Received HPV Test");
+
+        return cd;
+    }
+
+    /*Positive VIA or VIA/VILI result*/
+    public CohortDefinition positiveVIAresultCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i " +
+                "where (i.screening_method = 'VIA' or i.screening_method = 'VILI') and i.screening_result = 'Positive'" +
```
Comment (njorocs) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java:74]: consider using i.screening_method in ('','')

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+        return cd;
+    }
+
+    /*Treated Using Cryotherapy*/
+    public CohortDefinition treatedUsingCryotherapyCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i where i.screening_result = 'Positive'" +
+                " and i.treatment_method = 'Cryotherapy postponed' or i.treatment_method = 'Cryotherapy performed (single Visit)' " +
+                "or i.treatment_method = 'Cryotherapy performed' and visit_date between date(:startDate) and date(:endDate) group by i.patient_id;";
```
Comment (njorocs) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java:135]: consider i.treatment_method in(...)

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+        cd.setDescription("Other Treatment Given");
+
+        return cd;
+    }
+
+    /*HIV Positive Clients Screened*/
+    public CohortDefinition HIVPositiveClientsScreenedCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i inner join kenyaemr_etl.etl_hts_test t on t.patient_id = i.patient_id where t.final_test_result = 'Positive' and date(:startDate) and date(:endDate) group by i.patient_id;";
```
Comment (njorocs) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java:194]: 1. There are patients who will not be in hts_test coz they are transfer-ins (known positive). Get these from hiv enrolment. So you need to left join the two secondary tables (through sub-queries) with screening table and in the where clause check for those who are in either hts or hv enrolment
2. The date part has a bug. Which date are you comparing with the date parameters?

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+        cd.setDescription("HIV Positive Clients Screened");
+
+        return cd;
+    }
+
+    /*HIV Positive With Positive Screening Results*/
+    public CohortDefinition HIVPositiveClientsScreenedWithResultsCl() {
+        SqlCohortDefinition cd = new SqlCohortDefinition();
+        String sqlQuery = "select i.patient_id from kenyaemr_etl.etl_cervical_cancer_screening i inner join kenyaemr_etl.etl_hts_test t on t.patient_id = i.patient_id where t.final_test_result = 'Positive' and i.screening_result = 'Positive' and date(:startDate) and date(:endDate) group by i.patient_id;";
```
Comment (njorocs) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH745/Moh745CohortLibrary.java:208]: same as the previous comment

**Review by njorocs — COMMENTED** [review]
Also remove double blank lines where they appear

**Review by ojwanganto — APPROVED** [review]

**Review by njorocs — APPROVED** [review]

**Review by njorocs — APPROVED** [review]

**Review by njorocs — APPROVED** [review]

**Review by njorocs — APPROVED** [review]

**Pull request merged**